framework/src/dgkernels/DGKernelBase.C

```diff
@@ -109,53 +112,112 @@ DGKernelBase::DGKernelBase(const InputParameters & parameters)
     _save_in_strings(parameters.get<std::vector<AuxVariableName>>("save_in")),
     _diag_save_in_strings(parameters.get<std::vector<AuxVariableName>>("diag_save_in"))
 {
+  // Gather information on broken boundaries
+  auto & bnd = isParamValid("exclude_boundary")
+                   ? getParam<std::vector<BoundaryName>>("exclude_boundary")
+                   : std::vector<BoundaryName>(0);
+  auto bnd_ids = _mesh.getBoundaryIDs(bnd);
+
+  // check if the broken boundary ids are valid
+  auto & valid_ids = _mesh.meshSidesetIds();
+  std::vector<BoundaryID> diff;
+  std::set_difference(
```

Comment:
`std::set_difference` requires that the underlying container is sorted. If the user feeds in values in an unsorted manner you will not get the expected result here.

Reply:
Is there a recommended implementation? Should I be just looping through the given boundaries?

Reply:
Well it depends, you can load your two sets into a `std::set` then `set_difference` just works. Alternatively you can call std::sort on your vector before calling set_difference. You just have to make sure that the inputs are sorted.

Reply:
Thank you for the advice. I actually like the idea of using a loop, so that I can gather the actual names of wrongly inputted boundaries, instead of just the boundary IDs.